nexus/tests/config.test.toml

```diff
@@ -2,6 +2,9 @@
 # Oxide API: configuration file for test suite
 #
 
+# Avoid conflicting with other ports during HTTPS tests
```

Comment:
Why is this a separate top-level property?

Reply:
- We need to know what this is before the HTTPS server is running; it's possible that the certificate is only known after the "create_certificate" API is invoked
- In production, we want it to be 443
- If it's 443 for our tests, they'll fail, because they're binding to localhost and will conflict

Reply:
It's bizarre to pass `ConfigDropshot` here for the HTTPS server - if we [look at the struct](https://docs.rs/dropshot/0.8.0/dropshot/struct.ConfigDropshot.html), it contains the following:

```rust
pub struct ConfigDropshot {
    // Contains an address and port. Whatever address we pass ahead-of-time is a complete fabrication.
    pub bind_address: SocketAddr,
    // This part of the configuration is "true".
    pub request_body_max_bytes: usize,
    // This part of the configuration is a complete fabrication - when Nexus is launched, the certificates are not necessarily known.
    pub tls: Option<ConfigTls>,
}
```

So, basically: If we pass a `ConfigDropshot` for the external server before nexus has booted, it's only half valid, which IMO is confusing.

I'd rather pass configuration values that are "entirely true", and let Nexus fill in the rest (TLS info, addresses, etc) with known-at-runtime information.